vtkext/private/module/vtkF3DImguiActor.cxx

```diff
@@ -284,6 +284,89 @@ void vtkF3DImguiActor::RenderFileName()
   ImGui::End();
 }
 
+//----------------------------------------------------------------------------
+void vtkF3DImguiActor::RenderMetaData()
+{
+  ImGuiViewport* viewport = ImGui::GetMainViewport();
+
+  constexpr float marginRight = 5.f;
+
+  ImVec2 winSize = ImGui::CalcTextSize(this->MetaData.c_str());
+  winSize.x += 2.f * ImGui::GetStyle().WindowPadding.x;
+  winSize.y += 2.f * ImGui::GetStyle().WindowPadding.y;
+
+  // it's super important to set the size of the window manually
+  // otherwise ImGui skip a frame for computing the size resulting in
+  // no UI when doing offscreen rendering
+  ImGui::SetNextWindowSize(winSize);
+  ImGui::SetNextWindowPos(ImVec2(viewport->WorkSize.x - winSize.x - marginRight,
+    viewport->GetWorkCenter().y - 0.5f * winSize.y));
+  ImGui::SetNextWindowBgAlpha(0.35f);
+
+  ImGuiWindowFlags flags = ImGuiWindowFlags_NoDecoration | ImGuiWindowFlags_NoSavedSettings |
+    ImGuiWindowFlags_NoFocusOnAppearing | ImGuiWindowFlags_NoNav | ImGuiWindowFlags_NoMove;
+
+  ImGui::Begin("MetaData", nullptr, flags);
+  ImGui::TextUnformatted(this->MetaData.c_str());
+  ImGui::End();
```

Comment:
looks like you could factorize

Reply:
I could but I'd rather not because each widget will have a different code in the future when we'll work on the styling.

Reply:
right